### PR TITLE
Updated Packet++ headers to use `override` and Cpp type casts.

### PR DESCRIPTION
**Pull request by Dimi1010**
- Updated classes in Packet++ to use `override` specifier.
- Updated empty destructors to use `= default` instead of `{}`
- Updated many C casts to Cpp casts in Packet++ headers.
- Removed `missingOverrides` cppcheck suppression from everywhere except `Examples` and `Pcap++` as those are the only modules that currently raise `override` warnings.

### CODE REVIEW

File: Packet++/header/ArpLayer.h [hunk 1]
```diff
@@ -93,7 +92,7 @@ namespace pcpp
 		 */
 		inline arphdr* getArpHeader() const
 		{
-			return (arphdr*)m_Data;
+			return reinterpret_cast<arphdr*>(m_Data);
```
Comment (tigercosmos) [inline Packet++/header/ArpLayer.h:95]: Should be fine to use `static_cast ` here?

Comment (Dimi1010) [reply Packet++/header/ArpLayer.h:95]: Tried, does not work. `static_cast` between unrelated pointers works only if one of the types is a `void*`.

Comment (tigercosmos) [reply Packet++/header/ArpLayer.h:95]: I see. Thanks for trying.

**Review by tigercosmos — APPROVED** [review]
LGTM. Thanks for working on this tedious but important refactoring.

**Comment by clementperon** [comment]
LGTM but didn't check all the files, @Dimi1010 did you use CPPcheck to do this ?

**Comment by Dimi1010** [comment]
> LGTM but didn't check all the files, @Dimi1010 did you use CPPcheck to do this ?

Yes, when I remembered about it. Started by searching for them manually initially.

File: Packet++/header/IPSecLayer.h [hunk 2]
```diff
@@ -109,26 +109,26 @@ namespace pcpp
 		/**
 		 * @return The size of the AH header
 		 */
-		size_t getHeaderLen() const
+		size_t getHeaderLen() const override
 		{
-			return 4 * (getAHHeader()->payloadLen + 2);
+			return static_cast<size_t>(4) * (getAHHeader()->payloadLen + 2);
```
[the remark this answers is not in the document]
Comment (Dimi1010) [reply Packet++/header/IPSecLayer.h:114]: VS analysis gives me a "subexpression may overflow" notice without the cast.

File: Packet++/header/NdpLayer.h [hunk 3]
```diff
@@ -65,23 +64,23 @@ namespace pcpp
 
 		// implement abstract methods
 
-		size_t getTotalSize() const
+		size_t getTotalSize() const override
 		{
 			if (m_Data == nullptr)
-				return (size_t)0;
+				return 0u;  // Cpp23 - uz literal
```
[the remark this answers is not in the document]
Comment (Dimi1010) [reply Packet++/header/NdpLayer.h:70]: Sure, I guess.

Comment (Dimi1010) [inline Packet++/header/NdpLayer.h:70]:
```suggestion
				return 0;
```

File: Packet++/header/NdpLayer.h [hunk 4]
```diff
@@ -65,23 +64,23 @@ namespace pcpp
...
+			return static_cast<size_t>(m_Data->recordLen) * 8;
 		}
 
-		size_t getDataSize() const
+		size_t getDataSize() const override
 		{
 			if (m_Data == nullptr)
 			{
-				return 0;
+				return 0u;  // Cpp23 - uz literal
```
Comment (Dimi1010) [inline Packet++/header/NdpLayer.h:79]:
```suggestion
				return 0;
```

File: Packet++/header/NdpLayer.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dimi1010) [reply Packet++/header/NdpLayer.h]: Fixed 3472012

File: Packet++/header/IPSecLayer.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (seladb) [reply Packet++/header/IPSecLayer.h:114]: ok 👍

**Review by seladb — APPROVED** [review]

**Pull request merged**